Document Reactive Rest Client support of dynamic query parameters

michalvavrik opened the pull request:
Documents: [24783](https://github.com/quarkusio/quarkus/pull/24783)

Reactive Rest Client supports sending of query parameters passed as a `Map`, however I can't see this in documentation. There should be at least one reference to this feature.

michalvavrik commented:
@geoand WDYT?

geoand commented on docs/src/main/asciidoc/rest-client-reactive.adoc:
```diff
@@ -134,6 +134,44 @@ The `getById` method above is a blocking call. It should not be invoked on the e
+public interface ExtensionsService {
+
+    @GET
+    Set<Extension> getByIdAndName(@QueryParam("id") String id, @RestQuery String name); <1>
```
I don't think we want to mix both `@RestQuery` and `@QueryParam`

geoand reviewed (commented):
Thanks for this!

I added a couple comments

michalvavrik commented:
Thank you for the review.

geoand reviewed (approved):
Thanks!

michalvavrik committed:
Document Reactive Rest Client support of dynamic query parameters

Documents: [24783](https://github.com/quarkusio/quarkus/pull/24783)

Reactive Rest Client sending query paramters passed as a `Map`, however I can't see this in documentation. There should be at least reference to this feature.
